cmd/kubeadm/app/cmd/join.go

```diff
@@ -542,7 +533,7 @@ func (j *Join) BootstrapKubelet(tlsBootstrapCfg *clientcmdapi.Config) error {
 	}
 
 	// This feature is disabled by default in kubeadm
-	if features.Enabled(j.cfg.FeatureGates, features.DynamicKubeletConfig) {
+	if features.Enabled(j.initCfg.FeatureGates, features.DynamicKubeletConfig) {
```

Comment:
nit: in the future this should be use/be called `ClusterConfiguration`, because the thing returned from the cluster configmap at this stage is a clusterconfig, not an initconfig
This is not specific to this PR though

Reply:
Yes, I'll file a bug for that later, thank you.